yjit/src/core.rs

```diff
@@ -325,6 +325,15 @@ pub enum BranchShape {
 type BranchGenFn =
     fn(cb: &mut Assembler, target0: CodePtr, target1: Option<CodePtr>, shape: BranchShape) -> ();
 
+/// A place that a branch could jump to
+#[derive(Debug)]
+struct BranchTarget {
```

Comment:
Could `[repr(packed)]` work here?

Reply:
I just tried it but that gave me 12 errors that seem a bit hard to fix. Maybe in a separate PR.